frontend/public/components/factory/details.tsx

```diff
@@ -147,43 +154,45 @@ export const DetailsPage = withFallback<DetailsPageProps>(({ pages = [], ...prop
           kind={kindObj}
         />
       )}
-      <Firehose
-        resources={[...(_.isNil(props.obj) ? [objResource] : []), ...(props.resources ?? [])]}
-      >
-        <PageHeading
-          obj={props.obj}
-          detail={true}
-          title={props.title || props.name}
-          titleFunc={props.titleFunc}
-          menuActions={props.menuActions}
-          buttonActions={props.buttonActions}
-          customActionMenu={props.customActionMenu}
-          kind={props.customKind || props.kind}
-          breadcrumbs={pluginBreadcrumbs}
-          breadcrumbsFor={
-            props.breadcrumbsFor ??
-            (!pluginBreadcrumbs ? breadcrumbsForDetailsPage(kindObj, props.match) : undefined)
-          }
-          resourceKeys={resourceKeys}
-          getResourceStatus={props.getResourceStatus}
-          customData={props.customData}
-          badge={props.badge || getBadgeFromType(kindObj?.badge)}
-          icon={props.icon}
+      <PageTitleContext.Provider value={titleProviderValues}>
```

Comment:
You can replace the top level React fragment (`<>`) with this `PageTitleContext.Provider` to reduce the diff here.